File: app/src/main/java/easv/oe/dicecup2/DiceRollLog.java
package easv.oe.dicecup2;

import androidx.annotation.NonNull;

import java.util.ArrayList;

public class DiceRollLog {

    private ArrayList<Integer> dices;

    public DiceRollLog(ArrayList<Integer> dices) {
        this.dices = dices;
    }

    public int getDiceAmount(){
        return dices.size();
    }

    public String getDiceAmountString(){
        return "Amount of rolled dice: " + getDiceAmount();
    }

    public ArrayList<Integer> getDices() {
        return dices;
    }

    public String getRollInstance() {
        StringBuilder str = new StringBuilder("Amount of Dice: " + getDiceAmount() + " - Roll Instance: ");

        for (int i = 0; i< dices.size(); i++) {
            str.append(dices.get(i));
            if(i< dices.size()-1){
                str.append(" - ");
            }

        }
        return str.toString();
    }

    @NonNull
    @Override
    public String toString() {
        return getRollInstance();
    }
}

File: app/src/main/java/easv/oe/dicecup2/dice/DiceRollLog.kt
package easv.oe.dicecup2.dice

import java.io.Serializable
import java.text.SimpleDateFormat
import java.util.*

data class DiceRollLog(val dices: ArrayList<Int>) : Serializable {

    val date: Date = Calendar.getInstance().time;
    var format: SimpleDateFormat = SimpleDateFormat("dd-MM-yyyy HH:mm:ss")
    var formatted = format.format(date)

    val diceAmount: Int
        get() = dices.size

    val diceAmountString: String
        get() = "Amount of rolled dice: $diceAmount"

    val diceTimeString: String
        get() = "Time of Roll: $formatted"

    private val rollInstance: String
        get() {
            val str = StringBuilder("");
            for (i in dices.indices) {
                str.append(dices[i])
                if (i < dices.size - 1) {
                    str.append(" - ")
                }
            }
            return str.toString()
        }

    override fun toString(): String {
        return rollInstance
    }
}